hnn_core/parallel_backends.py

```diff
@@ -522,30 +551,30 @@ class MPIBackend(object):
     ----------
 
     n_procs : int
-        The number of processes MPI will actually use (spread over cores). This
-        can be less than the user specified value if limited by the cores on
-        the system, the number of cores allowed by the job scheduler, or
-        if mpi4py could not be loaded.
-    mpi_cmd: list of str
+        The number of processes MPI will actually use (spread over cores). If 1
+        is specified or mpi4py could not be loaded, the simulation will be run
+        with the JoblibBackend
+    mpi_cmd : list of str
         The mpi command with number of procs and options to be passed to Popen
     expected_data_length : int
         Used to check consistency between data that was sent and what
         MPIBackend received.
+    proc_queue : threading.Queue
+        A Queue object to hold process handles from Popen in a thread-safe way.
+        There will be a valid process handle present the queue when a MPI
+        åsimulation is running.
     """
     def __init__(self, n_procs=None, mpi_cmd='mpiexec'):
         self.expected_data_length = 0
+        self.proc = None
+        self.proc_queue = Queue()
 
         n_logical_cores = multiprocessing.cpu_count()
         if n_procs is None:
             self.n_procs = n_logical_cores
         else:
             self.n_procs = n_procs
 
-        # obey limits set by scheduler
-        if hasattr(os, 'sched_getaffinity'):
```

Comment:
why is this not needed anymore?

Reply:
Travis sets affinity to 2, but what we really want is to run on 3 cores for the oversubscribe test. This was most useful on a cluster, where you might be allocated 8 cores, but the machine would have 24 cores. Passing `n_procs=24` will start 24 processes. For that one use case, I think it's fine to make this the responsibility of the user. They will be hyper-aware of how many cores they are using, so it's a reasonable ask.

FYI, you can see justifications for changes in commit messages